File: javascript/isPrimeOrComposite.js
var inputData=process.argv;

if(inputData.length<3){
  return console.log("Please provide number as parameter");
}

for(var i=2; i<inputData.length;i++){
  var testData=inputData[i]
  var divideTill=parseInt(testData/2)
  if( isNaN( testData ) ){
    console.log(testData+ ": is Invalid Number");
  }else {
    if(testData==0 || testData==1){
      console.log(testData+ ": is neither Prime nor Composite Number");
    }else {
      if(testData==2 || testData==3){
        console.log(testData+ ": is  Prime Number");
      }else {
        for( var j=2; j<=divideTill; j++ ){
          if(testData%j == 0){
            console.log(testData+ ": is Composite Number");
            break;
          }
          if( j==divideTill ){
            console.log(testData+ ": is  Prime Number");
          }
        }
      }
    }
  }
}

File: java/isPrimeOrComposite.java
public class isPrimeOrComposite{
  public static void main(String[] args){
    String inputData[]=args;

    if(inputData.length<1){
      System.out.println("Please provide number as parameter");
      return ;
    }

    for(int i=0; i<inputData.length; i++){
      int testData=0;
      int divideTill=0;
      try{
      testData = Integer.parseInt(inputData[i]);
      divideTill=testData/2;
      }catch(NumberFormatException e){
        System.out.println(inputData[i]+ ": is Invalid Number");
        continue;
      }
      if(testData==0 || testData==1){
        System.out.println(testData+ ": is neither Prime nor Composite Number");
        continue;
      }
      if(testData==2 || testData==3){
        System.out.println(testData+ ": is  Prime Number");
        continue;
      }

      for( int j=2; j<=divideTill; j++ ){
          if( testData%j == 0 ){
            System.out.println(testData+ ": is Composite Number");
            break;
          }
          if( j==divideTill ){
            System.out.println(testData+ ": is Prime Number");
            break;
          }
      }

    }

  }
}
